include/modem/location.h

```diff
@@ -0,0 +1,326 @@
+	 */
+	uint16_t timeout;
+
+	/** Used Wi-Fi positioning service. */
+	enum location_service service;
+};
+
+/** Positioning method configuration. */
+struct location_method_config {
+	/** Positioning method. */
+	enum location_method method;
+	union {
+		/** Configuration for LOCATION_METHOD_CELLULAR. */
+		struct location_cellular_config cellular;
+		/** Configuration for LOCATION_METHOD_GNSS. */
+		struct location_gnss_config gnss;
+		/** Configuration for LOCATION_METHOD_WIFI. */
+		struct location_wifi_config wifi;
+	};
+};
+
+/** Location request configuration. */
+struct location_config {
+	/** Number of location methods in 'methods'. */
+	uint8_t methods_count;
+	/**
+	 * @brief Selected positioning methods and associated configurations in priority order.
+	 *
+	 * @details Index 0 has the highest priority. Number of methods is indicated in
+	 * 'methods_count'.
+	 */
+	struct location_method_config *methods;
+	/**
+	 * @brief Position update interval in seconds.
+	 *
+	 * @details Set to 0 for a single position update. For periodic position updates
+	 * the valid range is 10...65535 seconds.
+	 */
+	uint16_t interval;
+};
+
+/**
+ * @brief Event handler prototype.
+ *
+ * @param[in] event_data Event data.
+ */
+typedef void (*location_event_handler_t)(const struct location_event_data *event_data);
+
+/**
+ * @brief Initializes the library.
+ *
+ * @details Initializes the library and sets the event handler function.
+ * This can be called multiple times, which sets the event handler again.
+ *
+ * @param[in] event_handler Event handler function.
+ *
+ * @return 0 on success, or negative error code on failure.
+ */
+int location_init(location_event_handler_t event_handler);
+
+/**
+ * @brief Requests the current position or starts periodic position updates.
+ *
+ * @details Requests the current position using the given configuration. Depending on the
+ * configuration, a single position or periodic position updates are given. If the position
+ * request is successful, the results are delivered to the event handler function in
+ * LOCATION_EVT_LOCATION event. Otherwise LOCATION_EVT_TIMEOUT or LOCATION_EVT_ERROR event is
+ * triggered.
+ *
+ * In periodic mode, position update is always attempted after the configured interval
+ * regardless of the outcome of any previous attempt. Periodic position updates can be
+ * stopped by calling location_cancel().
+ *
+ * @param[in] config Used configuration or NULL to get a single position update with
+ *                   the default configuration.
+ *
+ * @return 0 on success, or negative error code on failure.
+ */
+int location_request(const struct location_config *config);
+
+/**
+ * @brief Cancels periodic position updates.
+ *
+ * @return 0 on success, or negative error code on failure.
+ */
+int location_request_cancel(void);
+
+/**
+ * @brief Sets default values to given configuration.
+ *
+ * @details Given methods are set as a list of methods for the given configuration.
```

Comment:
Added a sentence about this. I still wanted to keep memory reservation comments to clearly tell the responsibilities.
This may all be removed if we end up changing methods to a static table.